tests/integration/test_core_organizer_batch3.py

```
@@ -0,0 +1,1876 @@
+    def test_get_duplicate_space_calculates_correctly(self, tmp_path: Path) -> None:
+        from file_organizer.services.analytics.storage_analyzer import StorageAnalyzer
+
+        f = tmp_path / "dup.bin"
+        f.write_bytes(b"x" * 500)
+
+        groups = [{"files": [str(f), str(tmp_path / "copy1.bin"), str(tmp_path / "copy2.bin")]}]
+
+        analyzer = StorageAnalyzer()
+        wasted = analyzer.get_duplicate_space(groups)
+
+        assert wasted == 1000  # 2 extra copies × 500 bytes
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)